sys/net/gnrc/routing/rpl/gnrc_rpl.c

```diff
@@ -245,10 +252,13 @@ static void _parent_timeout(gnrc_rpl_parent_t *parent)
 static void *_event_loop(void *args)
 {
     msg_t msg, reply;
+    mutex_t *eventloop_startup = (mutex_t*)args;
 
-    (void)args;
     msg_init_queue(_msg_q, GNRC_RPL_MSG_QUEUE_SIZE);
 
+    /* Message queue is initialized, gnrc_rpl_init can continue */
+    mutex_unlock(eventloop_startup);
```

Comment:
Can we maybe put `eventloop_startup` into its own block, so the memory is freed again and no one gets the stupid idea to reuse it, once unlocked (since the mutex is allocated on the stack of the initializing thread it is not allocated anymore, once the caller function is left)?

Reply:
It's not changing anything in the built code (the compiler just passes on args, probably), but it does enhance visibility, and I've extended the comment around the unlock as an extra precaution.

Reply:
Yeah, I was thinking it might be optimized away anyway and was more concerned about the visibility of that variable :-).